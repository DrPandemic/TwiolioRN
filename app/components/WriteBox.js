// @flow

import React, { Component } from 'react';
import {
  Dimensions,
  StyleSheet,
  View,
  TextInput,
} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Icon } from 'react-native-elements';

import { ActionCreators } from '../actions';

const screenWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 0,
    height: 80,
    zIndex: 2,
    flexDirection: 'row',
  },
  input: {
    height: 50,
    width: screenWidth,
    marginRight: -40,
  },
  icon: {
    height: 50,
    paddingTop: 10,
    paddingRight: 15,
  },
});

type PropsT = {
  sendMessage: (string, string, string) => void,
  to: string,
  from: string,
}

export class PWriteBox extends Component {
  props: PropsT;

  state: {
    text: string,
  }

  constructor(props: PropsT) {
    super(props);
    this.state = { text: '' };

    this.send = this.send.bind(this);
  }

  send() {
    this.props.sendMessage(this.props.to, this.props.from, this.state.text);
    this.setState({ text: '' });
  }

  renderInput() {
    return (
      <TextInput
        style={styles.input}
        returnKeyType={'send'}
        onChangeText={text => this.setState({ text })}
        placeholder={'Type an SMS message'}
        onSubmitEditing={() => { this.send(); }}
        value={this.state.text}
      />
    );
  }

  renderButton() {
    return (
      <Icon
        style={styles.icon}
        name='send'
        color='#aaa'
        underlayColor='#00000000'
        raised={true}
        onPress={() => { this.send(); }}
      />
    );
  }

  render() {
    return (
      <View style={styles.container}>
        {this.renderInput()}
        {this.renderButton()}
      </View>
    );
  }
}

function mapStateToProps() {
  return {};
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(ActionCreators, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(PWriteBox);
