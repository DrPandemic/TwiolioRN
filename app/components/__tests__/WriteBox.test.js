// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import { PWriteBox } from '../WriteBox';

test('renders a the box', () => {
  const box = renderer.create(
    <PWriteBox
      to='foo'
      from='bar'
    />
  ).toJSON();

  expect(box).toMatchSnapshot();
});

test('can send from button', () => {
  const spy = jest.fn();
  const box = new PWriteBox({
    to: 'foo',
    from: 'bar',
    sendMessage: spy,
  });
  box.state.text = 'baz';

  const renderedButton = box.renderButton();
  renderedButton.props.onPress();

  expect(spy).toBeCalledWith('foo', 'bar', 'baz');
});

test('can send from keyboard', () => {
  const spy = jest.fn();
  const box = new PWriteBox({
    to: 'foo',
    from: 'bar',
    sendMessage: spy,
  });
  box.state.text = 'baz';

  const renderedButton = box.renderInput();
  renderedButton.props.onSubmitEditing();

  expect(spy).toBeCalledWith('foo', 'bar', 'baz');
});
