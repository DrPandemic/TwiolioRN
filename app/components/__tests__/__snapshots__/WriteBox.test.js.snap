// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a the box 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "flexDirection": "row",
      "height": 80,
      "position": "absolute",
      "zIndex": 2,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Type an SMS message"
    returnKeyType="send"
    style={
      Object {
        "height": 50,
        "marginRight": -40,
        "width": 750,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "height": 50,
          "paddingRight": 15,
          "paddingTop": 10,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#aaa",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î…£
    </Text>
  </View>
</View>
`;
